packages/server-wallet/package.json

```diff
@@ -101,7 +101,8 @@
     "test:chain": "jest -c ./jest/jest.chain.config.js --runInBand",
     "test:ci": "yarn test --bail && yarn test:chain --bail",
     "test:e2e": "./src/test_db.sh server_wallet_payer server_wallet_receiver; yarn tsc; yarn jest -c ./jest/jest.e2e.config.js --runInBand",
-    "test:stress": "./src/test_db.sh server_wallet_payer server_wallet_receiver; yarn tsc; NODE_ENV=test npx ts-node ./e2e-test/scripts/stress-test.ts"
+    "test:stress": "./src/test_db.sh server_wallet_payer server_wallet_receiver; yarn tsc; NODE_ENV=test npx ts-node ./e2e-test/scripts/stress-test.ts",
+    "test:with-peers": "yarn jest -c ./jest/jest.with-peers.config.js --runInBand"
```

Comment:
I see this script added here, but it's not reference elsewhere.

A natural extension of this work is
1. don't run the `with-peers` tests when running `yarn test`
2. Add a new step to circle to run `yarn test:with-peers`.

This would shorten the total length of the CircleCI run by maybe 2 minutes, at the cost of an additional 30-60s startup cost for the step introduced in (2).

Reply:
The other thing to note is that if the `Shortcomings` section in the description is addressed and `beforeAll`/`afterAll` are included in the jest config, `yarn test` which maps to `./jest/jest.config.js` will no longer work with peer tests.

Reply:
@andrewgordstewart My expectation of `yarn test` in a package would be that it runs all tests of that package - would the suggestion from (1) break that?

Reply:
At the moment, `yarn test` does not run all tests in a package. It only runs tests that use `./jest/jest.config.js`

Reply:
I'll clarify a bit more.
1. As you know, a `jest.config.js` file can specify which tests should run with that config. Eg. `./jest/jest.config.js` uses [this `testMatch`](https://github.com/statechannels/statechannels/blob/2cb32c512e700c788224cec28896d2258904d1b0/packages/server-wallet/jest/jest.config.js#L8)
2. There's a few other configs [here](https://github.com/statechannels/statechannels/tree/master/packages/server-wallet/jest). You've added one in this PR.
3. CircleCI runs some of these tests in different steps. EG. the `test:e2e` script is run in the [`server-wallet-e2e-test`](https://app.circleci.com/pipelines/github/statechannels/statechannels/12046/workflows/30438ae2-7758-4e24-ab81-1cec41a09645/jobs/58746) job, while the `test` script is run in the [`test`](https://app.circleci.com/pipelines/github/statechannels/statechannels/12046/workflows/30438ae2-7758-4e24-ab81-1cec41a09645/jobs/58745) job.


Expanding on (3) a bit, the `test` job runs [`yarn test:ci`](https://app.circleci.com/pipelines/github/statechannels/statechannels/12046/workflows/30438ae2-7758-4e24-ab81-1cec41a09645/jobs/58745/parallel-runs/0/steps/0-106) _from the monorepo root_.

There is a lot of value in making the `test` job run faster, by running the `test:with-peers` test cases into a different circle step. Therefore, my suggestion is to
1. do not run the `with-peers` step in `test:ci`
2. Create a new circle step which runs `test:with-peers` OR run `test:with-peers` after `test:e2e` OR a sensible alternative.

This would lower the start-to-finish duration of the CI run by ~2 minutes, expecting that number to grow over time.

... Back to your comment:
> My expectation of `yarn test` in a package would be that it runs all tests of that package 

As it turns out, we can **and maybe should** change `yarn test` to run all the tests in the package, as a local way of quickly "running all the tests". It might do this by running `yarn run-s test:ci test:e2e test:with-peers test:stress ...`, or by running some shell script.